### PR TITLE
Migration composant lidar

### PR DESCRIPTION
**Pull request by llienher**

### CODE REVIEW

File: src/download/service.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import {saveAs} from 'file-saver';
+
+/**
+ * A function to start a download for a file.
+ *
+ * @param {string} content The content of the file to download.
+ * @param {string} fileName The name of the file to download.
+ * @param {string | undefined} opt_fileType The type of the file to download.
+ */
+export function download(content: string, fileName: string, opt_fileType: string | undefined): void {
```
Comment (sbrunner) [inline src/download/service.ts:32]: Why not default export?

Comment (llienher) [reply src/download/service.ts:32]: What is the benefit of calling this something else when you want to use it ? :/

Comment (llienher) [reply src/download/service.ts:32]: Yes, thanks. :-)

File: src/controllers/vars_desktop.scss [hunk 2]
```diff
@@ -59,4 +59,9 @@ $notification-width: 12.5rem !default;
   --footer-panel-height: 12.5rem;
   --footer-panel-height-lidarresult: 21.88rem;
   --current-footer-panel-height: 0;
+
+  // LIDAR profile
+  --app-margin: 0.62rem;
+  --lidarprofile-height: 21.88rem;
+  --profile-legend-width: 9.38rem;
```
Comment (sbrunner) [inline src/controllers/vars_desktop.scss:66]: --footer-panel-height-lidarresult  should be in this section why --app-margin is in this section?

Comment (llienher) [reply src/controllers/vars_desktop.scss:66]: the CSS I added used this variable, I wanted to keep the code as similar as the initial one for the beginning.

Comment (sbrunner) [reply src/controllers/vars_desktop.scss:66]: Yes, but it's a common variable already defined as SCSS variable then it should be 
```
--app-margin: #{$app-margin};
```
some lines above :-)

File: src/controllers/vars_desktop.scss [hunk 3]
```diff
@@ -57,6 +57,11 @@ $notification-width: 12.5rem !default;
...
   --current-left-panel-border-width: #{$border};
   --footer-panel-height: 12.5rem;
-  --footer-panel-height-lidarresult: 21.88rem;
+  --footer-panel-height-lidar: 21.88rem;
   --current-footer-panel-height: 0;
+
+  // LIDAR profile
+  --app-margin: #{$app-margin};
+  --lidarprofile-height: 21.88rem;
+  --profile-legend-width: 9.38rem;
```
Comment (sbrunner) [inline src/controllers/vars_desktop.scss:66]:
```suggestion
  --current-footer-panel-height: 0;
  --app-margin: #{$app-margin};

  // LIDAR profile
  --footer-panel-height-lidar: 21.88rem;
  --lidarprofile-height: 21.88rem;
  --profile-legend-width: 9.38rem;
```
Better organisation :-)

File: src/controllers/vars_desktop.scss (hunk 2 above)
Comment (sbrunner) [reply src/controllers/vars_desktop.scss:66]: Why don't you want to put `--footer-panel-height-lidar` after the LIDAR comment?

Comment (sbrunner) [reply src/controllers/vars_desktop.scss:66]: Thanks :-)

File: srcapi/store/line.ts [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+   *
+   * @param line The line
+   */
+  setLine(line: OlGeomLineString): void {
+    this.line_.next(line);
+  }
+}
+
+const line = new LineModel();
+export default line;
```
Comment (sbrunner) [inline srcapi/store/line.ts:67]: Should be moved to lidar...

File: src/lidar/Manager.ts [hunk 5]
```diff
@@ -277,7 +277,11 @@ export class LidarprofileManager {
       this.isPlotSetup_ = false;
     }
 
-    d3select('#gmf-lidarprofile-container .lidar-error').style('visibility', 'hidden');
+    const lidarContainerElement = document
+      .querySelector('gmf-desktop-canvas gmf-lidar-footer')
```
Comment (ger-benjamin) [inline src/lidar/Manager.ts:281]:
```suggestion
      .querySelector('#gmf-lidar-footer')
```
I would use an id here. That would limit "not element found" errors.
Same below, do a git grep please

File: src/map/FeatureOverlayMgr.ts [hunk 6]
```diff
@@ -0,0 +1,191 @@
...
+  /**
+   * @param {OlMap} map Map.
+   */
+  init(map: OlMap): void {
+    this.layer_.setMap(map);
+  }
+
+  /**
+   * @param {OlStyleStyleStyleLike} style
+   * Style.
```
Comment (ger-benjamin) [inline src/map/FeatureOverlayMgr.ts:164]: I would like to see more comment, but this one is too much
```suggestion
```

**Review by ger-benjamin — COMMENTED** [review]
Not tested, but looks good

**Pull request merged**